source/common/http/conn_manager_impl.cc

```diff
@@ -349,8 +347,10 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
       *request_headers_, connection_manager_.stats_.prefix_, connection_manager_.stats_.store_);
 
   // Make sure we are getting a codec version we support.
-  const HeaderString& codec_version = request_headers_->Version()->value();
-  if (!(codec_version == "HTTP/1.1" || codec_version == "HTTP/2")) {
+  Protocol protocol = connection_manager_.codec_->protocol();
+  if (!(protocol == Protocol::Http11 || protocol == Protocol::Http2)) {
```

Comment:
Just in case we add something else in the future, but the chance of that happening is basicaly zero so I will fix it.